pkg/ride/tree_evaluation_test.go

```diff
@@ -2568,29 +2568,10 @@ func TestInvokeDAppFromDAppScript6(t *testing.T) {
 	NewWrappedSt := initWrappedState(smartState(), env, tree.LibVersion)
 	wrappedSt = *NewWrappedSt
 
-	res, err := CallFunction(env, tree, "foo", proto.Arguments{})
-
-	require.NoError(t, err)
-	r, ok := res.(DAppResult)
-	require.True(t, ok)
-
-	sr, ap, err := proto.NewScriptResult(r.actions, proto.ScriptErrorMessage{})
-	require.NoError(t, err)
-	assert.Equal(t, 0, len(ap))
-	expectedActionsResult := &proto.ScriptResult{
-		DataEntries:  make([]*proto.DataEntryScriptAction, 0),
-		Transfers:    make([]*proto.TransferScriptAction, 0),
-		Issues:       make([]*proto.IssueScriptAction, 0),
-		Reissues:     make([]*proto.ReissueScriptAction, 0),
-		Burns:        make([]*proto.BurnScriptAction, 0),
-		Sponsorships: make([]*proto.SponsorshipScriptAction, 0),
-		Leases:       make([]*proto.LeaseScriptAction, 0),
-		LeaseCancels: make([]*proto.LeaseCancelScriptAction, 0),
-	}
-	assert.Equal(t, expectedActionsResult, sr)
+	_, err = CallFunction(env, tree, "foo", proto.Arguments{})
```

Comment:
Why the function with a recursive call should return an error? I see you changed the number of invocation count, but how is that related with a recursive call?

Reply:
Because it should return error in case of exceeding recursion limit